base/compiler/ssair/passes.jl

```diff
@@ -1239,10 +1239,45 @@ function type_lift_pass!(ir::IRCode)
     ir
 end
 
+function is_bb_empty(ir::IRCode, bb::BasicBlock)
+    isempty(bb.stmts) && return true
+    if length(bb.stmts) == 1
+        stmt = ir[SSAValue(first(bb.stmts))][:inst]
+        return stmt === nothing || isa(stmt, GotoNode)
+    end
+    return false
+end
+
+# TODO: This is terrible, we should change the IR for GotoIfNot to gain an else case
```

Comment:
I would be in favor of removing fall-through for terminators

Reply:
Yes, me too, but the reason they're there is that the interpreter doesn't have a notion of BasicBlocks. However, we can forbid them in IRCode certainly.